Signal/src/call/CallService.swift

```diff
@@ -428,6 +427,9 @@ protocol CallServiceObserver: class {
         let backgroundTask = UIApplication.shared.beginBackgroundTask {
             let timeout = CallError.timeout(description: "background task time ran out before call connected.")
             DispatchQueue.main.async {
+                guard self.call == newCall else {
```

Comment:
There are many places where we were responding to events related to obsolete calls.  I've added a bunch of `guard` statements that finally fix a number of nagging issues (i.e. unexpected PCC).